planning/behavior_path_planner/src/scene_module/lane_change/normal.cpp

```diff
@@ -567,19 +567,26 @@ std::vector<double> NormalLaneChange::sampleLongitudinalAccValues(
   return utils::lane_change::getAccelerationValues(min_acc, max_acc, longitudinal_acc_sampling_num);
 }
 
-double NormalLaneChange::calcPrepareDuration(
+std::vector<double> NormalLaneChange::calcPrepareDuration(
   const lanelet::ConstLanelets & current_lanes, const lanelet::ConstLanelets & target_lanes) const
 {
   const auto & common_parameters = planner_data_->parameters;
-  const auto threshold = lane_change_parameters_->min_length_for_turn_signal_activation;
-  const auto current_vel = getEgoVelocity();
+  const auto base_link2front = planner_data_->parameters.base_link2front;
+  const auto threshold =
+    lane_change_parameters_->min_length_for_turn_signal_activation + base_link2front;
 
-  // if the ego vehicle is close to the end of the lane at a low speed
-  if (isNearEndOfCurrentLanes(current_lanes, target_lanes, threshold) && current_vel < 1.0) {
-    return 0.0;
+  std::vector<double> prepare_durations;
+  constexpr double step = 0.5;
+
+  for (double duration = common_parameters.lane_change_prepare_duration; duration >= 0.0;
+       duration -= step) {
+    prepare_durations.push_back(duration);
+    if (!isNearEndOfCurrentLanes(current_lanes, target_lanes, threshold)) {
+      break;
+    }
   }
 
-  return common_parameters.lane_change_prepare_duration;
+  return prepare_durations;
```

Comment:
If it regulates lane change on crosswalk or intersection, lane change path may be found if the prepare duration is longer.
So, it returns some prepare duration candidates.

Reply:
I can roughly imagine why the prepare_duration would be shortened when the vehicle is moving at low speed and is near the end of the lane. Is there any degradation in performance if this feature is lost? Since the duration is being explored starting from zero, it seems like there wouldn't be an issue, but I'm curious if the low-speed condition is really necessary.

Reply:
Sorry exploring from `common_parameters.lane_change_prepare_duration`...